Sources/_CryptoExtras/SPX/SPX_boring.swift

```diff
@@ -0,0 +1,176 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCrypto open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftCrypto project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_implementationOnly import CCryptoBoringSSL
+@_implementationOnly import CCryptoBoringSSLShims
+import Crypto
+@_implementationOnly import CryptoBoringWrapper
+import Foundation
+
+#if swift(>=5.8)
+@_documentation(visibility: public)
+public enum SPX {}
+#else
+public enum SPX {}
+#endif
+
+extension SPX {
+    public struct PrivateKey: Sendable {
+        private let pointer: UnsafeMutablePointer<UInt8>
+        
+        public init() {
+            self.pointer = UnsafeMutablePointer<UInt8>.allocate(capacity: 64)
+            CCryptoBoringSSL_spx_generate_key(UnsafeMutablePointer<UInt8>.allocate(capacity: 32), self.pointer)
+        }
+        
+        public init(from seed: some DataProtocol) throws {
+            guard seed.count >= (3 * 16) else {
```

Comment:
It would probably be beneficial with a `public static let giveMeASuitableName = 3 * 16` (`seedLength`?) and replace a bunch multiplication expressions with that.